File: exercises/week02/src/solid/MotionSensor.java
package solid;

import java.util.Random;

/**
 * A security sensor for detecting motion.
 */
public class MotionSensor implements Sensor {

    private String location;
    private Random random;

    public MotionSensor(String location) {
        this.location = location;
        random = new Random();
    }

    @Override
    public boolean isTriggered() {
        return random.nextInt(100) < 30;
    }

    @Override
    public String getLocation() {
        return location;
    }

    @Override
    public String getSensorType() {
        return "Motion sensor";
    }

    @Override
    public Type getType() {
        return Type.SECURITY;
    }
}

File: exercises/week02/src/solid/MotionSensor.scala
package solid

/**
  * A sensor which is not battery-powered and is a security sensor (unlike SmokeSensor and FireSensor)
  */
class MotionSensor extends Sensor {
  override def isTriggered: Boolean = false

  override def getLocation: String = "unknown location"

  override def getSensorType: String = "motion sensor"

  override def getSensorCategory = Category.Security
}
